pkgs/development/compilers/ghc/cabal-paths.patch

```diff
@@ -0,0 +1,40 @@
+diff --git a/Cabal/Distribution/Simple/Build/PathsModule.hs b/Cabal/Distribution/Simple/Build/PathsModule.hs
+index 5e660e8d6..22ce5b4fe 100644
+--- a/libraries/Cabal/Cabal/Distribution/Simple/Build/PathsModule.hs
++++ b/libraries/Cabal/Cabal/Distribution/Simple/Build/PathsModule.hs
+@@ -32,6 +32,7 @@ import Distribution.Pretty
+ import Distribution.Version
+ 
+ import System.FilePath ( pathSeparator )
++import Data.List ( isPrefixOf )
+ 
+ -- ------------------------------------------------------------
+ -- * Building Paths_<pkg>.hs
+@@ -82,8 +83,8 @@ generatePathsModule pkg_descr lbi clbi =
+         pragmas++
+         "module " ++ prettyShow paths_modulename ++ " (\n"++
```

Comment:
A big concern is that the error messages relating to this will be *super* confusing. Sure enough, it won't fail if what you are doing is okay, but if you try run into a case that is problematic it'll be quite confusing.

Sadly there doesn't seem to be a pragma for removed exports, so maybe adding a `WARNING` to the module is the best we can do? This will be quite spammy though.

Reply:
I've added a `WARNING` for each function that gets deleted by this patch.  If a function should be omitted, instead of deleting it completely, I'm just returning the empty string instead, that allows me to write out the `WARNING`s on a per-function basis.

Reply:
Did you forget to push? In any case I don't think that's such a good idea, since it means that it will show warnings, but still compile successfully (we can't really tick on `-Werror` for everything), leading to incorrect runtime results.